python/pyiceberg/types.py

```diff
@@ -232,31 +238,40 @@ class NestedField(IcebergType):
 
     field_id: int = Field(alias="id")
     name: str = Field()
-    field_type: IcebergType = Field(alias="type")
+    field_type: IcebergSubTypes = Field(alias="type")
     required: bool = Field(default=True)
     doc: Optional[str] = Field(default=None, repr=False)
-    initial_default: Any = Field(alias="initial-default", repr=False)
+    initial_default: Optional[Any] = Field(alias="initial-default", default=None, repr=False)
 
     def __init__(
         self,
         field_id: Optional[int] = None,
         name: Optional[str] = None,
-        field_type: Optional[IcebergType] = None,
+        field_type: Optional[Type[IcebergType]] = None,
```

Comment:
What is happening here? Isn't the type an instance of `IcebergType`?

Reply:
This is incorrect indeed,